2024 08 09 speakers page

avivzk2 opened the pull request.

Commit:
Added speakers generator from pretalx api

Commit:
Added speakers page generation to CI

Commit:
Added speakers page template

Commit:
Updates to CI

Commit:
Fixed missing speakers

Commit:
Updates to remove speakers that will agree to publish

shaib commented on .github/workflows/deploy.yml:
```diff
@@ -70,6 +70,11 @@ jobs:
+      - name: Generated spakers page
+        run: |
+          pip install requests
+          python speakers.py ${{ vars.PRETALX_EVENT_NAME }} -t ${{ vars.PRETALX_API_TOKEN }} -o website_2024/content/pages/speakers.md --verbose
```
Don't you want the API token to be a secret, rather than a var?
I mean, for now, the detailed logs of action runs are not publicly available, so you're only sharing your token with people who have some privileged access to the repo -- it's not critical. But still...

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+    speakers_list = []
+    for speaker in speakers:
+        avatar = speaker['avatar'] if speaker['avatar'].strip() else './theme/img/PyConIL.png'
+        role = speaker['answers']['role'] if 'role' in speaker['answers'] else ''
+        company = speaker['answers']['company'] if 'company' in speaker['answers'] else ''
```
```suggestion
        role = speaker['answers'].get('role', '')
        company = speaker['answers'].get('company', '')
```

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+        social_links = []
+        for social, value in speaker['answers'].items():
+            if social in SOCIAL_ICONS:
+                stype = SOCIAL_ICONS[social][0]
+                url = SOCIAL_ICONS[social][1](value)
```
I think this would be more readable as
```suggestion
                stype, url_fixer = SOCIAL_ICONS[social]
                url = url_fixer(value)
```
and I would consider renaming `stype` to `icon_class`

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Get PyCon speakers as csv')
```
```suggestion
    parser = argparse.ArgumentParser(description='Generate PyCon speakers page')
```

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+    parser.add_argument('event', type=str, help="Event name (pretalx slug)")
+    parser.add_argument('-t', '--token', required=True, help="API token")
+    parser.add_argument('-o', '--output', type=str, help="Output speakers md file")
+    parser.add_argument('-f', '--force', action='store_true', help="Force download")
```
Seems to be unused

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+    return data
+
+
+def filter_speakers(speakers, talks):
```
Ironically, according to https://docs.pretalx.org/api/resources/speakers/#get--api-events-event-speakers- if you use an unauthenticated request -- no token -- you already get the list of "confirmed and publicly scheduled" speakers, and there's no need to filter by talks.

(though I'm not sure with that request you'll get all the answers)

shaib commented on website_2024/themes/PyCon-Israel-Flex/templates/speakers.html:
```diff
@@ -0,0 +1,31 @@
+{% block title %} &ndash; {{ page.title|striptags|escape }}{% endblock %}
+
+{% block content %}
+<article class="single" style="max-width: 80%;">
```
I note that the only difference between this and the `page.html` template used for most other pages is the addition of `style="max-width: 80%;"` in this line.

There is probably a less-self-repeating way to do that, but it makes me ask -- is this maybe a change we want to include for all pages?

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+            avatar=avatar,
+            name=speaker['name'].title(),
+            role=f"{role} @ {company}",
+            bio=speaker['biography'],
```
```suggestion
            bio=speaker['biography'] or '',
```

to prevent this: 
![image](https://github.com/user-attachments/assets/73fef9dd-6724-42e6-bbf7-d657bb17d8cd)

shaib commented on speakers.py:
```diff
@@ -0,0 +1,321 @@
+Template: speakers
+save_as: speakers.html
+Lang: en
+URL:
+page_number: 7
```
I'm not sure what the empty URL setting is doing, and I think the speakers page should be after the agenda
(which 15).
```suggestion
page_number: 17
```
At some point, we should do a "renumber" to make the numbers 10 apart again

shaib reviewed (commented):
Thanks, this looks very good, although it still needs some fixes.

The specific comments below are mostly things which I think would improve this, but not crucial.

But there are display issues which need to be addressed -- when the role (combined role and company) is two lines (happens a lot), and the bio is more than three lines (which is almost always the case), the bio hides the social links.
![image](https://github.com/user-attachments/assets/b2b64e0a-48df-4ae3-a9b7-66e320e587ea)

I suggest replacing `{role} @ {company}` with `{role} <br/> {company}`, to make it always two lines, and then fixing the CSS (make speaker cell taller, or bio shorter).

One more issue I noted -- in your beta site, this page shows up only in English. In `pelicanconf.py` we have a setting `I18N_UNTRANSLATED_PAGES = 'keep'` which is supposed to make it show up there too. Do you know why it doesn't show up?

shaib committed:
Minor easy changes per suggestions on PR

shaib commented:
I'm starting to look at fixing along the lines I drew above

shaib committed:
Fix speaker image croppings by making speaker images square

shaib committed:
Make speaker biography a bit shorter before hover

To make more room for speaker social links

shaib committed:
Add requests to dependencies instead of in deployment

Requests is needed for performing API calls to Pretalx
Currently, to get speakers info

shaib committed:
Make reference to PyCon logo (missing speaker image) more robust

This way it also survives in the page in the Hebrew site
